libraries/AC_Avoidance/AP_OAPathPlanner.cpp

```diff
@@ -180,15 +212,31 @@ AP_OAPathPlanner::OA_RetState AP_OAPathPlanner::mission_avoidance(const Location
 void AP_OAPathPlanner::avoidance_thread()
 {
     while (true) {
+#if !HAL_MINIMIZE_FEATURES
```

Comment:
It's not a blocker but I don't think the HAL_MINIMIZE_FEATURES should apply to only the object database.  I think we should disable all of the object avoidance.  The way it is here, bendyruler will avoid the fence but not dynamic obstacles.  I don't think having only half the feature isn't a good idea.

Reply:
very much agreed. It was silly to place the requirement on just this part of the memory hungry avoidance. We can include the whole avoidance thing in HAL_MINIMIZE_FEATURES in another PR

Reply:
great.  I've created an issue here so we don't forget about it. https://github.com/ArduPilot/ardupilot/issues/11770